plugins/node/opentelemetry-instrumentation-pg/src/utils.ts

```diff
@@ -161,7 +169,35 @@ export function handleInvalidQuery(
   return result;
 }
 
+export function handleExecutionResult(
+  config: PgInstrumentationConfig,
+  span: Span,
+  pgResult: pgTypes.QueryResult | pgTypes.QueryArrayResult | unknown
+) {
+  if (
+    config.enhancedDatabaseReporting &&
+    config.responseHook !== undefined &&
+    pgResult !== undefined
+  ) {
+    safeExecuteInTheMiddle(
+      () => {
+        config.responseHook!(
+          span,
+          pgResult as pgTypes.QueryResult | pgTypes.QueryArrayResult
+        );
+      },
+      err => {
+        if (err) {
+          diag.error('Error running response hook', err);
```

Comment:
Great idea, I’d suggest implementing this (in a separate PR, of course) at the InstrumentationBase level so that it won't be necessary to manually add the prefix separately for each instrumentation class. WDYT?

Reply:
I think it's a great idea :)

Reply:
I now read the v0.21.0 release notes and looks like it was added there:
https://github.com/open-telemetry/opentelemetry-js/pull/2261

Reply:
It was indeed added on instrumentation but you should use `this._diag` from the instrumentation class to be able to use it. I think you'll need to give the instance class here to make it work

Reply:
@vmarchaud, @blumamir for this PR I recommend not adding this change. 
`_diag` is a [protected member](https://github.com/obecny/opentelemetry-js/blob/ee3bcccc8852f6570d29e0edeffad8899b54e87c/packages/opentelemetry-instrumentation/src/instrumentation.ts#L38), it's accessible only internally within the class or any class that extends it but not externally. Since the pg instrumentation uses the `utils` module, passing the instrumentation will not be enough and a refactor is needed here.

Reply:
Yes, I agree. We tried to refactor our instrumentations as well and bumped into the same issue, and decided not to use this component logger after all.
However, we made sure all the log prints are prefixed with `${packageName} instrumentation:`, [for example](https://github.com/aspecto-io/opentelemetry-ext-js/pull/130/files#diff-d03f0c16859a12d3ca9f6763f0c3acb05f00c16f1cabbb65f25ee941674017d1R108).

But I see there is no convention for the contrib instrumentations on that, so it's up to you.

Reply:
why not passing the component logger to the util class directly from class when you call the util method ?